src/test/java/org/opensearch/knn/index/util/LuceneTests.java

```diff
@@ -108,4 +108,9 @@ public void testSetInitialized() {
         luceneLibrary.setInitialized(true);
         assertTrue(luceneLibrary.isInitialized());
     }
+
+    public void testVersion() {
+        Lucene luceneInstance = Lucene.INSTANCE;
+        assertEquals("9.5.0", luceneInstance.getVersion());
```

Comment:
This test will fail when OS upgrades to a new version. Let's not hard code this version.  See if we can get this value from some other constant which is present in OS. So that when that constants updates the test uses the latest version.

I think you can use ```Version.LATEST.toString()```

Reply:
Yes, it will fail on OS upgrade, I actually done it on purpose for us to make sure version upgrade doesn't flow unattended, like it was with 9.2 that made all the way up to 9.5. With variables it's harder to check which exact value it has.

Reply:
But here is the problem, if we have hard code the value as "9.5.0" as intentional, then in src file too we should make it 9.5.0 rather than Version.LATEST.toString(). Because the next time this Unit test will fail and then we need to take a decision whether to upgrade or not.

Reply:
I think we need to discuss if k-NN engine is different from codec. For codec now we may have 9.4 (through bwc package) but lucene library will be 9.5. Does it mean the whole lucene k-NN engine is 9.4 as well? Engine is kNN concept, we can change it as needed.
We use codec, reader, writer, format that are essentially coming from codec. But rest of classes are from current lucene version (9.5 in my example).

My opinion - version of codec and engine should match. IMO we should upgrade to latest codec from our lucene version (like 9.5 that is in 2.5 snapshot). But if can keep older codec versions then engine should remain 9.4 too.
@jmazanec15 what do you think?

Reply:
Here is our current guidance on Codec versioning: https://github.com/opensearch-project/k-NN/blob/main/DEVELOPER_GUIDE.md#codec-versioning.

Right now, we have 2 versioning systems - the first marks the codec version used for lucene, the second marks the version for k-NN. This is a bit confusing, however, I believe that the reasoning is so that we can properly read older segment files.

Reply:
@martin-gaievski on doing some deep-dive I understand the purpose of this code change better. We are updating the version of lucene, which is not related to codec over here.

> My opinion - version of codec and engine should match. IMO we should upgrade to latest codec from our lucene version (like 9.5 that is in 2.5 snapshot). But if can keep older codec versions then engine should remain 9.4 too.

I agree that we should upgrade the codec version to include the latest codec. I will pick up that change. 

But still my first comment hold over here. In the test we should not hard code string as "9.5.0", we should get this from `Version.LATEST.toString()`. Because this is unit test we want to make sure that if someone changes the value to other version which is not latest, in the src files tests should detect it.

Reply:
ack, let me update to Version.LATEST